superset-frontend/src/features/alerts/AlertReportModal.test.tsx

```diff
@@ -17,68 +17,640 @@
+        last_name: 'Admin',
+      },
+    ],
+    recipients: [
+      {
+        id: 1,
+        recipient_config_json: '{"target": "test@user.com"}',
+        type: 'Email',
+      },
+    ],
+    report_format: 'PNG',
+    sql: 'Select * From DB',
+    timezone: 'America/Rainy_River',
+    type: 'Alert',
+    validator_config_json: '{"threshold": 10.0, "op": "<"}',
+    validator_type: 'operator',
+    working_timeout: 3600,
+  };
+  if (dashboard) {
+    return {
+      ...mockPayload,
+      dashboard: { id: 1, dashboard_title: 'Test Dashboard' },
+    };
+  }
+  return {
+    ...mockPayload,
+    chart: {
+      id: 1,
+      slice_name: 'Test Chart',
+      viz_type: 'table',
+    },
+  };
+};
+
+// mocking resource endpoints
+const FETCH_DASHBOARD_ENDPOINT = 'glob:*/api/v1/report/1';
+const FETCH_CHART_ENDPOINT = 'glob:*/api/v1/report/2';
+
+fetchMock.get(FETCH_DASHBOARD_ENDPOINT, { result: generateMockPayload(true) });
+fetchMock.get(FETCH_CHART_ENDPOINT, { result: generateMockPayload(false) });
+
+// Related mocks
+const ownersEndpoint = 'glob:*/api/v1/alert/related/owners?*';
+const databaseEndpoint = 'glob:*/api/v1/alert/related/database?*';
+const dashboardEndpoint = 'glob:*/api/v1/alert/related/dashboard?*';
+const chartEndpoint = 'glob:*/api/v1/alert/related/chart?*';
+
+fetchMock.get(ownersEndpoint, { result: [] });
+fetchMock.get(databaseEndpoint, { result: [] });
+fetchMock.get(dashboardEndpoint, { result: [] });
+fetchMock.get(chartEndpoint, { result: [{ text: 'table chart', value: 1 }] });
+
+// Create a valid alert with all required fields entered for validation check
+
+// @ts-ignore will add id in factory function
+const validAlert: AlertObject = {
+  active: false,
+  changed_on_delta_humanized: 'now',
+  created_on: '2023-12-12T22:33:25.927764',
+  creation_method: 'alerts_reports',
+  crontab: '0 0 * * *',
+  dashboard_id: 0,
+  chart_id: 0,
+  force_screenshot: false,
+  last_state: 'Not triggered',
+  name: 'Test Alert',
+  owners: [
+    {
+      first_name: 'Superset',
+      id: 1,
+      last_name: 'Admin',
+    },
+  ],
+  recipients: [
+    {
+      type: 'Email',
+      recipient_config_json: { target: 'test@user.com' },
+    },
+  ],
+  timezone: 'America/Rainy_River',
+  type: 'Alert',
+};
+
+jest.mock('../buildErrorTooltipMessage', () => ({
+  buildErrorTooltipMessage: jest.fn(),
+}));
+
+const generateMockedProps = (
+  isReport = false,
+  useValidAlert = false,
+  useDashboard = true,
+): AlertReportModalProps => {
+  let alert;
+  // switching ids for endpoint when testing dashboard vs chart edits
+  if (useDashboard) {
+    alert = { ...validAlert, id: 1 };
+  } else {
+    alert = { ...validAlert, id: 2 };
+  }
+
+  return {
+    addDangerToast: () => {},
+    addSuccessToast: () => {},
+    onAdd: jest.fn(() => []),
+    onHide: jest.fn(),
+    alert: useValidAlert ? alert : null,
+    show: true,
+    isReport,
+  };
+};
+
+// combobox selector for mocking user input
+const comboboxSelect = async (
```

Comment:
This is ok but generally speaking tests are meant to be dumb. We should not be creating logics inside tests